src/Compilers/CSharp/Portable/Symbols/Source/SourceNamespaceSymbol.AliasesAndUsings.cs

```diff
@@ -198,27 +221,175 @@ public ImmutableArray<NamespaceOrTypeAndUsingDirective> GetUsingNamespacesOrType
                     throw ExceptionUtilities.UnexpectedValue(declarationSyntax);
             }
 
-            foreach (var declaration in _mergedDeclaration.Declarations)
+            return GetAliasesAndUsings(declarationSyntax).GetUsingNamespacesOrTypes(this, declarationSyntax, basesBeingResolved);
+        }
+
+        private Imports GetGlobalUsingImports(ConsList<TypeSymbol>? basesBeingResolved)
+        {
+            return GetMergedGlobalAliasesAndUsings(basesBeingResolved).Imports;
+        }
+
+        private ImmutableDictionary<string, AliasAndUsingDirective> GetGlobalUsingAliasesMap(ConsList<TypeSymbol>? basesBeingResolved)
+        {
+            return GetMergedGlobalAliasesAndUsings(basesBeingResolved).UsingAliasesMap!;
+        }
+
+        private ImmutableArray<NamespaceOrTypeAndUsingDirective> GetGlobalUsingNamespacesOrTypes(ConsList<TypeSymbol>? basesBeingResolved)
+        {
+            return GetMergedGlobalAliasesAndUsings(basesBeingResolved).UsingNamespacesOrTypes;
+        }
+
+        private MergedGlobalAliasesAndUsings GetMergedGlobalAliasesAndUsings(ConsList<TypeSymbol>? basesBeingResolved, CancellationToken cancellationToken = default)
+        {
+            if (_lazyMergedGlobalAliasesAndUsings is null)
```

Comment:
nit: consider reducing the depth of this method by using an `if (_lazyMergedGlobalAliasesAndUsings is not null) { return _lazyMergedGlobalAliasesAndUsings; }` quick exit at the top.